include/real_time_tools/threadsafe/threadsafe_timeseries.hpp

```diff
@@ -14,11 +14,13 @@
 
 #include <memory>
 #include <vector>
+#include <atomic>
+#include <mutex>
```

Comment:
I know it is a bit (too) late to say this but, why not using the real_time api of posix? We are designing a real-time tool here no?
Therefore I would have expected to see the real_time_tools/thread.hpp to be included.

This said, I think the std is also fine, I am not sure how terrible it is to use std instead of posix here.

Reply:
ah yes, this is actually a good point. we should probably test how much difference this makes.

so max, this means in real-time code we should never use std mutexes etc?

Reply:
This is something that needs benchmarking or research. I am not sure how much we gain by using the real-time api of the os on this. But there is a xenomai and a posix implementation of the mutexes and everything. So I guess this makes sense at some point.

So my naive answer is: yes